code/controllers/CMSMain.php

```diff
@@ -1316,7 +1316,6 @@ public function duplicate($request) {
 		
 		if(($id = $this->urlParams['ID']) && is_numeric($id)) {
 			$page = DataObject::get_by_id("SiteTree", $id);
-			if($page && (!$page->canEdit() || !$page->canCreate())) return Security::permissionFailure($this);
```

Comment:
Logic being:
A user cannot duplicate a page they can't see.
A user cannot duplicate to a place they can't create.
A user can still edit a page they've been given access to.